app/src/main/java/com/example/internetshop/data/repository/ProductsCategoryRepositoryImpl.kt

```diff
@@ -13,13 +12,12 @@ import javax.inject.Inject
 class ProductsCategoryRepositoryImpl @Inject constructor(
     private val categoryApi: CategoryApi,
     private val productMapper: ProductMapper,
-    private val productServerMapper: ProductServerMapper
+    private val productServerMapper: ProductServerMapper,
+    private val categoryRequestMapper: CategoryRequestMapper
 ) : ProductsCategoryRepository {
-    override fun getProductsCategory(categoryRequest: CategoryRequest): Single<List<SimpleProduct>> {
+    override fun getProductsCategory(categoryName: String): Single<List<SimpleProduct>> {
         return categoryApi.getCategoryProducts(
-            categoryRequest.getCategoryName() ?: throw NoSuchCategoryException(
-                "Category name cannot be null"
-            )
+            categoryRequestMapper.toRequest(categoryName).categoryName
```

Comment:
For which purpose you create request model with `categoryName` as parameter and then use `model.categoryName` as argument for api call? Why don't you use `categoryName` straightaway? 
You can write something like `categoryAPi.getCategoryProducts(categoryName)`.

Reply:
fixed